src/foam/nanos/crunch/Capability.js

```diff
@@ -282,6 +284,65 @@ foam.CLASS({
 
       return today.after(capabilityExpiry);
       `
+    },
+    {
+      name: 'hasPrerequisitesSatisfied',
+      type: 'Boolean',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'ucjDaoKey', type: 'String' },
+        { name: 'ucjUserPredicate', type: 'Predicate' },
+      ],
+      documentation: `
+        Check if this capability is satisfied with its prerequisites.
+        Default implementation requires all prerequisites.
+      `,
+      javaCode: `
+        // Get a list of prerequisite capabilities
```

Comment:
Should we not be adjusting the logic for the UserCapabilityJunctionDAO decorator with this PR?

Then I have to wonder when you are iterating over the UCJ's here - we then are duplicating the iteration in the above named decorator.

I'd like to see how this all plays out with the changes to ucjDAO

Reply:
> Should we not be adjusting the logic for the UserCapabilityJunctionDAO decorator

This would be replacing the similar code in UserCapabilityJunctionDAO (well, UserCapabilityJunctionDAO would be calling on this code). The reason doing it here instead of there would be that now it's the subclass of Capability's job to determine how its prerequisites are satisfied.

The reason I didn't remove the code in UserCapabilityJunctionDAO in this PR was I was hoping it could get merged as-is and then we have some opportunity to play with the idea and see if we run into hurdles.